liana-gui/src/app/export.rs

```diff
@@ -0,0 +1,275 @@
+    }
+}
+
+impl From<std::io::Error> for Error {
+    fn from(value: std::io::Error) -> Self {
+        Error::Io(format!("{:?}", value))
+    }
+}
+
+impl From<DaemonError> for Error {
+    fn from(value: DaemonError) -> Self {
+        Error::Daemon(format!("{:?}", value))
+    }
+}
+
+#[derive(Debug)]
+pub enum State {
+    Init,
+    Running,
+    Stopped,
+}
+
+#[derive(Debug, Clone)]
+pub enum ExportProgress {
+    Started(Arc<Mutex<JoinHandle<()>>>),
+    Progress(f32),
+    Ended(Box<PathBuf>),
+    Finnished,
+    Error(Error),
+    None,
+}
+
+pub struct ExportState {
+    pub receiver: Receiver<ExportProgress>,
+    pub sender: Option<Sender<ExportProgress>>,
+    pub handle: Option<Arc<Mutex<JoinHandle<()>>>>,
+    pub daemon: Arc<dyn Daemon + Sync + Send>,
+    pub path: Box<PathBuf>,
+}
+
+impl ExportState {
+    pub fn new(daemon: Arc<dyn Daemon + Sync + Send>, path: Box<PathBuf>) -> Self {
+        let (sender, receiver) = channel();
+        ExportState {
+            receiver,
+            sender: Some(sender),
+            handle: None,
+            daemon,
+            path,
+        }
+    }
+
+    pub async fn start(&mut self) {
+        if let (true, Some(sender)) = (self.handle.is_none(), self.sender.take()) {
+            let daemon = self.daemon.clone();
+            let path = self.path.clone();
+
+            let cloned_sender = sender.clone();
+            let handle = tokio::spawn(async move {
+                let dir = match path.parent() {
+                    Some(dir) => dir,
+                    None => {
+                        send_error!(sender, NoParentDir);
+                        return;
+                    }
+                };
+                if !dir.exists() {
+                    if let Err(e) = fs::create_dir_all(dir) {
+                        send_error!(sender, e.into());
+                        return;
+                    }
+                }
+                let mut file = match File::create(path.as_path()) {
+                    Ok(f) => f,
+                    Err(e) => {
+                        send_error!(sender, e.into());
+                        return;
+                    }
+                };
+                let header = "Date,Label,Value,Fee,Txid,Block\n".to_string();
+                if let Err(e) = file.write_all(header.as_bytes()) {
+                    send_error!(sender, e.into());
+                    return;
+                }
+
+                let info = daemon.get_info().await;
+
+                let start = match info {
+                    Ok(info) => info.timestamp,
+                    Err(e) => {
+                        send_error!(sender, e.into());
+                        return;
+                    }
+                };
+                // look 2 hour forward
+                let end = ((Utc::now() + Duration::hours(2)).timestamp()) as u32;
+
+                let history = daemon.list_history_txs(start, end, u32::MAX as u64).await;
```

Comment:
remote backend api have limits, you will need to paginate this instead of batch fetch all of it.

Reply:
what is the limit? shouldn't limits be enforced by the list_history_txs() API?

Reply:
You can use 20, this is the default limit.